src/renderer/modules/components/index.ts

```diff
@@ -0,0 +1,59 @@
+import { ModuleExports } from "../../../types";
+import { error } from "../logger";
+
+const modulePromises: Array<Promise<void>> = [];
+
+function importTimeout<T extends ModuleExports>(
+  name: string,
+  moduleImport: Promise<T>,
+  cb: (mod: T) => void,
+): void {
+  modulePromises.push(
+    new Promise<void>((res, rej) => {
+      const timeout = setTimeout(() => {
+        error("Replugged", "Components", void 0, `Could not find module "${name}"`);
+        rej(new Error(`Module not found: "${name}`));
+      }, 10_000);
```

Comment:
Does this need to be 10 seconds? That's a long time.

Reply:
5 seconds wasn't very consistent so i did 10

Reply:
Fair enough